clippy_lints/src/unused_io_amount.rs

```diff
@@ -67,27 +67,38 @@ impl<'tcx> LateLintPass<'tcx> for UnusedIoAmount {
 }
 
 fn check_method_call(cx: &LateContext<'_>, call: &hir::Expr<'_>, expr: &hir::Expr<'_>) {
-    if let hir::ExprKind::MethodCall(path, _, _, _) = call.kind {
-        let symbol = &*path.ident.as_str();
-        let read_trait = match_trait_method(cx, call, &paths::IO_READ);
-        let write_trait = match_trait_method(cx, call, &paths::IO_WRITE);
+    if let hir::ExprKind::MethodCall(ref path, _, ref args, _) = call.kind {
+        match &*path.ident.as_str() {
+            "ok" | "or" | "or_else" => {
+                check_method_call(cx, &args[0], expr);
```

Comment:
There should be tests for any added cases (`or`, `or_else`).

This recursion is pretty odd. It seems like these method cases should be handled near the same place where `unwrap` is handled. You could have a loop there to traverse method calls.

Reply:
Ok :) just that I can't place the check on `ok`, `or` and `or_else` at the same place as `unwrap` since we need to detect `reader.read(&mut result).ok()?;` with the `?` other white it does not detect it.